torch/distributed/_shard/partial_tensor.py

```diff
@@ -252,16 +273,27 @@ def partial_cat(types, args=(), kwargs=None):
         if not isinstance(input, _PartialTensor):
             raise RuntimeError('All inputs need to be an instance of _PartialTensor')
         if idx == 0:
-            reduce_op = input.reduce_op
-        elif reduce_op != input.reduce_op:
-            raise RuntimeError('All _PartialTensor reduce_ops need to be the same, found: {reduce_op} and {input.reduce_op}')
+            reduce_op = input._reduce_op
+        elif reduce_op != input._reduce_op:
+            raise RuntimeError(
+                'All _PartialTensor reduce_ops need to be the same, found: '
+                '{reduce_op} and {input._reduce_op}'
+            )
 
-        local_shards.append(input.local_shard)
+        local_shards.append(input._local_shard)
 
     if kwargs is None:
         dim = 0
     else:
         if 'out' in kwargs:
             raise RuntimeError('"out" kwarg is not supported!')
         dim = kwargs['dim'] if 'dim' in kwargs else 0
-    return _PartialTensor(torch.cat(local_shards, dim), input._process_group, input.reduce_op)
+
+    return _PartialTensor(torch.cat(local_shards, dim), input._process_group, input._reduce_op)
+
+@_custom_partial_tensor_op(torch.Tensor.size)
```

Comment:
Maybe we can do it in follow up PRs. Is this the only tensor property op that we need to dispatch? i.e. what about `shape`, `dtype`, `layout`, `dim`

Reply:
Yeah, I'll add these and consolidate with `register_default_op` in a separate PR.